crates/cubecl-macros/src/lib.rs

```diff
@@ -197,3 +201,55 @@ pub fn derive_autotune_key(input: TokenStream) -> TokenStream {
         Err(e) => e.into_compile_error().into(),
     }
 }
+
+/// *Internal macro*
+///
+/// Generates an implementation of `OperationArgs` for this type. All fields must implement
+/// `FromArgList`.
+#[doc(hidden)]
+#[proc_macro_derive(OperationArgs, attributes(args))]
+pub fn derive_operation_args(input: TokenStream) -> TokenStream {
+    let input = syn::parse(input).unwrap();
+    match generate_op_args(input) {
+        Ok(tokens) => tokens.into(),
+        Err(e) => e.into_compile_error().into(),
+    }
+}
+
+/// *Internal Macro*
+///
+/// Generates reflection info for an operation. Generates an opcode enum and an implementation of
+/// `OperationReflect` that deconstructs and reconstructs the typed version. All variant fields must
+/// implement `OperationArgs`, or `OperationReflect` if the variant is nested. Uses the `operation`
+/// helper attribute.
+///
+/// # Arguments
+///
+/// * `opcode_name` - the name of the generated opcode enum (required)
+/// * `pure` - marks this entire operation as pure
+/// * `commutative` - marks this entire operation as commutative
+///
+/// # Variant arguments
+///
+/// * `pure` - Marks this variant as pure
+/// * `commutative` - Marks this variant as commutative
+///
+#[doc(hidden)]
+#[proc_macro_derive(OperationReflect, attributes(operation))]
+pub fn derive_operation(input: TokenStream) -> TokenStream {
```

Comment:
I think it would be cleaner to create another crate project for the internal stuff related to the cubecl-ir operations.

Reply:
I went back and forth on that, so if you think so then I guess that tilts the scales to a separate crate. Should it just be nested under the IR crate or still top level?